DM-41256: Use ResourcePath to read FITS headers

timj opened the pull request:
This allows metadata translation to run on a FITS file in a S3 bucket.

timj committed:
Update pre-commit versions

timj committed:
Fix numpydoc problem

timj committed:
Update how license is specified

timj committed:
Use ResourcePath for reading headers

This allows metadata translation to run for FITS files in S3 buckets.

timj committed:
Fix type of parameter in test

timj committed:
Drop python 3.10

timj committed:
Update action versions

timj committed:
Do not continually upload to PyPI if nothing has changed

timj committed:
Move doc build to separate action

timj committed:
Suppress sphinx warning about ResourcePath et al.

Not entirely clear why intersphinx can't find the docstrings
in the pipelines.lsst.io docs.

timj committed:
Change one test to explicitly use a URI

timj committed:
Fix docstrings to refer to ResourcePath

timj committed:
Now depends on lsst.resources

mxk62 commented on python/astro_metadata_translator/file_helpers.py:
```diff
@@ -72,66 +62,117 @@ def _read_fits_metadata(file: str, hdu: int, can_raise: bool = False) -> Mutable
+        The requested header. `None` if it could not be read and
+        ``can_raise`` is `False`.
+    """
+    # For detailed docstrings see the _read_fits_metadata implementation below.
```
Unless I'm missing some context, it doesn't look like the docstring of `_read_fits_metadata()` below contains more information than this one. I'd remove either this comment or the docstring.

mxk62 commented on python/astro_metadata_translator/translators/decam.py:
```diff
@@ -438,7 +440,7 @@ def determine_translatable_headers(
         Parameters
         ----------
-        filename : `str`
+        filename : `str` or or `lsst.resources.ResourcePathExpression`
```
Too many "or"s, please fix.

timj committed:
Add python 3.14 to matrix

timj committed:
Add dependabot file

timj committed:
Disable scipy intersphinx mapping

Old documenteer uses it but it no longer exists in the expected
place and we don't need it anyhow.

timj committed:
Declare works with python 3.14
